front/src/modules/timesStops/TimesStops.tsx

```diff
@@ -186,18 +209,21 @@ const TimesStops = ({ pathProperties, pathSteps = [] }: TimesStopsProps) => {
         }
       }}
       stickyRightColumn={{
-        component: ({ rowData }) =>
+        component: ({ rowData, rowIndex }) =>
           createDeleteViaButton({
             removeVia: () => removeVia(rowData),
-            isRowVia: isVia(compact(pathSteps), rowData),
+            isRowVia:
+              rowIndex !== 0 &&
+              rowIndex !== pathProperties.allWaypoints?.length - 1 &&
+              isVia(compact(pathSteps), rowData, true),
```

Comment:
Hmm that would require that I put a lot more parameters in the function (rowIndex, pathProperties, pathSteps, rowData).

If you don't like the logic directly in the function argument, I could do like this : 
```
component: ({ rowData, rowIndex }) => {
          const isRowVia =
            rowIndex !== 0 &&
            rowIndex !== pathProperties.allWaypoints?.length - 1 &&
            isVia(compact(pathSteps), rowData, true);
          return createDeleteViaButton({
            removeVia: () => removeVia(rowData),
            isRowVia,
          });
        },
```

What do you prefer ?

Reply:
I understand that this significantly increases the number of parameters for the function, but I believe it's better to extract the logic into a smaller function. Initially, I suggested this because I found the component already quite difficult to read, and I thought that moving this test into a separate function would simplify the component's code.

What do you think ?

Reply:
Ok I understand. Done.